From fd28133cd4efcaa0d8d7a554ec4762ead9db31c7 Mon Sep 17 00:00:00 2001
From: Zhang Sanshan <b51434@freescale.com>
Date: Fri, 3 Apr 2015 14:56:18 +0800
Subject: [PATCH 468/602] MA-6511 when we enable GPU preload(+    # setprop
 ro.zygote.disable_gl_preload true), the
 board(sabresd dq) can't boot up.

1 add selinux policy for zygote.
2 Magd will write the fsl_magd.txt into /data/misc which contains the correcting data.
  The correcting data help to recognize the direction and orientation.

Signed-off-by: Zhang Sanshan <b51434@freescale.com>
---
 evk_6sl/BoardConfig.mk       | 3 ++-
 imx6/sepolicy/sensors.te     | 1 +
 imx6/sepolicy/zygote.te      | 1 +
 sabreauto_6q/BoardConfig.mk  | 3 ++-
 sabreauto_6sx/BoardConfig.mk | 3 ++-
 sabresd_6dq/BoardConfig.mk   | 3 ++-
 sabresd_6sx/BoardConfig.mk   | 3 ++-
 7 files changed, 12 insertions(+), 5 deletions(-)
 create mode 100644 imx6/sepolicy/zygote.te

diff --git a/evk_6sl/BoardConfig.mk b/evk_6sl/BoardConfig.mk
index e8d9c9a..46fae75 100644
--- a/evk_6sl/BoardConfig.mk
+++ b/evk_6sl/BoardConfig.mk
@@ -107,4 +107,5 @@ BOARD_SEPOLICY_UNION := \
        bootanim.te \
        dnsmasq.te \
        recovery.te \
-       device.te
+       device.te \
+       zygote.te
diff --git a/imx6/sepolicy/sensors.te b/imx6/sepolicy/sensors.te
index e19953e..4d34a7a 100644
--- a/imx6/sepolicy/sensors.te
+++ b/imx6/sepolicy/sensors.te
@@ -25,3 +25,4 @@ wakelock_use(sensors)
 allow sensors cgroup:dir { create add_name };
 allow sensors input_device:chr_file rw_file_perms;
 allow sensors input_device:dir r_dir_perms;
+allow sensors system_data_file:dir { write };
diff --git a/imx6/sepolicy/zygote.te b/imx6/sepolicy/zygote.te
new file mode 100644
index 0000000..a13c51b
--- /dev/null
+++ b/imx6/sepolicy/zygote.te
@@ -0,0 +1 @@
+allow zygote gpu_device:chr_file { read write open ioctl };
diff --git a/sabreauto_6q/BoardConfig.mk b/sabreauto_6q/BoardConfig.mk
index 6f5ed2e..6bc2d72 100644
--- a/sabreauto_6q/BoardConfig.mk
+++ b/sabreauto_6q/BoardConfig.mk
@@ -140,5 +140,6 @@ BOARD_SEPOLICY_UNION := \
        dnsmasq.te \
        recovery.te \
        install_recovery.te \
-       device.te
+       device.te \
+       zygote.te
 
diff --git a/sabreauto_6sx/BoardConfig.mk b/sabreauto_6sx/BoardConfig.mk
index 06e87f4..2858b43 100755
--- a/sabreauto_6sx/BoardConfig.mk
+++ b/sabreauto_6sx/BoardConfig.mk
@@ -130,4 +130,5 @@ BOARD_SEPOLICY_UNION := \
        dnsmasq.te \
        recovery.te \
        install_recovery.te \
-       device.te
+       device.te \
+       zygote.te
diff --git a/sabresd_6dq/BoardConfig.mk b/sabresd_6dq/BoardConfig.mk
index 9c81c93..1073047 100644
--- a/sabresd_6dq/BoardConfig.mk
+++ b/sabresd_6dq/BoardConfig.mk
@@ -147,6 +147,7 @@ BOARD_SEPOLICY_UNION := \
        bootanim.te \
        dnsmasq.te \
        recovery.te \
-       device.te
+       device.te \
+       zygote.te
 
 
diff --git a/sabresd_6sx/BoardConfig.mk b/sabresd_6sx/BoardConfig.mk
index 82f85ee..88bb387 100644
--- a/sabresd_6sx/BoardConfig.mk
+++ b/sabresd_6sx/BoardConfig.mk
@@ -134,4 +134,5 @@ BOARD_SEPOLICY_UNION := \
        bootanim.te \
        dnsmasq.te \
        recovery.te \
-       device.te
+       device.te \
+       zygote.te
-- 
1.8.0

